minecraft-codev-forge/codev-forge-runtime/src/main/java/net/msrandom/minecraftcodev/forge/mappings/CodevNameMappingService.java

```diff
@@ -0,0 +1,125 @@
+package net.msrandom.minecraftcodev.forge.mappings;
+
+import com.google.common.base.Suppliers;
+import cpw.mods.modlauncher.api.INameMappingService;
+import net.fabricmc.mappingio.format.tiny.Tiny2FileReader;
+import net.fabricmc.mappingio.tree.MappingTree;
+import net.fabricmc.mappingio.tree.MemoryMappingTree;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+
+public class CodevNameMappingService implements INameMappingService {
+    private static final String MAPPINGS_PATH_PROPERTY = "codev.naming.mappingsPath";
+
+    private final Supplier<Mappings> mappings = Suppliers.memoize(() -> {
+        Path path = Paths.get(Objects.requireNonNull(
+            System.getProperty(MAPPINGS_PATH_PROPERTY),
+            "Missing mappings path"
+        ));
+        InputStream mappingsStream;
+        try {
+            mappingsStream = Files.newInputStream(path);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Won't be null since we're requiring a property
+//        if (mappingsStream == null) {
+//            return new Mappings(Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap());
+//        }
```

Comment:
Are we need the null check instead of error?

Reply:
Nope, this null check is not needed anymore. Should probably be removed